bareRFM69.cpp

```diff
@@ -1,131 +1,172 @@
-/*
- *  This file is part of plainRFM69.
+/**
+ *
+ *  bareRFM69.cpp - A component of plainRFM69, created by Ivor Wanders.
+ *
  *  Copyright (c) 2014, Ivor Wanders
- *  MIT License, see the LICENSE.md file in the root folder.
-*/
+ *  Copyright (c) 2018, Charles Taylor
+ *
+ *  MIT License - See the LICENSE.md file for details.
+ *
+ **/
+
 
 #include "bareRFM69.h"
 
-// Most functions are implemented in the header file.
 
-void inline bareRFM69::chipSelect(bool enable){
-    digitalWrite(this->cs_pin, (enable) ? LOW : HIGH );
-}
+// ****************************************************
+// *** NOTE: Some functions are implemented and all ***
+// ***       are documented in the header file.     ***
+// ****************************************************
+
+
+// ********************************
+// * Register/FIFO read and write *
+// ********************************
+
+
+/**
+ *
+ *  These functions all perform the following progression:
+ *
+ *    1. Gain control of the SPI bus using acquireSPI().
+ *    2. Assert the chip select pin using acquireSPI().
+ *    3. Perform the SPI transfer.
+ *    4. Deassert the chip select pin using releaseSPI().
+ *    5. Release the SPI bus using releaseSPI().
+ *
+ **/
 
-void bareRFM69::writeRegister(uint8_t reg, uint8_t data){
-    SPI.beginTransaction(SPISettings(10000000, MSBFIRST, SPI_MODE0));  // gain control of SPI bus
-    this->chipSelect(true); // assert chip select
-    SPI.transfer(RFM69_WRITE_REG_MASK | (reg & RFM69_READ_REG_MASK)); 
-    SPI.transfer(data);
-    this->chipSelect(false);// deassert chip select
-    SPI.endTransaction();    // release the SPI bus
-}
 
-uint8_t bareRFM69::readRegister(uint8_t reg){
-    uint8_t foo;
-    SPI.beginTransaction(SPISettings(10000000, MSBFIRST, SPI_MODE0));  // gain control of SPI bus
-    this->chipSelect(true); // assert chip select
+uint8_t bareRFM69::readRegister(uint8_t reg)
+{
+    uint8_t data;
+
+    bareRFM69::acquireSPI();
```

Comment:
Shouldn't need the class `bareRFM69` here... if it isn't, someone could make it virtual and overload it from a subclass. Rule of thumb; if unnecessary don't specify the class. Same goes for the releaseSPI() here and in the other methods.